test/unit/snowflake_config_test.js

```diff
@@ -0,0 +1,215 @@
+        name: 'invalid ocspMode',
+        options: { ocspFailOpen: 'unsupported' },
+        errorCode: ErrorCodes.ERR_GLOBAL_CONFIGURE_INVALID_OCSP_MODE
+      },
+      {
+        name: 'invalid json parser',
+        options: { jsonColumnVariantParser: 'unsupported' },
+        errorCode: ErrorCodes.ERR_GLOBAL_CONFIGURE_INVALID_JSON_PARSER
+      },
+      {
+        name: 'invalid xml parser',
+        options: { xmlColumnVariantParser: 'unsupported' },
+        errorCode: ErrorCodes.ERR_GLOBAL_CONFIGURE_INVALID_XML_PARSER
+      },
+    ];
+
+  var createNegativeITCallback = function (testCase)
+  {
+    return function ()
+    {
+      var error;
+
+      try
+      {
+        snowflake.configure(testCase.options);
+      }
+      catch (err)
+      {
+        error = err;
+      }
+      finally
+      {
+        assert.ok(error);
+        assert.strictEqual(error.code, testCase.errorCode);
+      }
+    };
+  };
+
+  var index, length, testCase;
+  for (index = 0, length = negativeTestCases.length; index < length; index++)
+  {
+    testCase = negativeTestCases[index];
+    it(testCase.name, createNegativeITCallback(testCase));
+  }
+
+  ///////////////////////////////////////////////////////////////////////////
+  //// Test valid arguments                                              ////
+  ///////////////////////////////////////////////////////////////////////////
+
+  var testCases =
+    [
+      {
+        name: 'logLevel error',
+        options:
+        {
+          logLevel: LOG_LEVEL_TAGS.ERROR
+        }
+      },
+      {
+        name: 'logLevel warn',
+        options:
+        {
+          logLevel: LOG_LEVEL_TAGS.WARN
+        }
+      },
+      {
+        name: 'logLevel debug',
+        options:
+        {
+          logLevel: LOG_LEVEL_TAGS.DEBUG
+        }
+      },
+      {
+        name: 'logLevel info',
+        options:
+        {
+          logLevel: LOG_LEVEL_TAGS.INFO
+        }
+      },
+      {
+        name: 'logLevel trace',
+        options:
+        {
+          logLevel: LOG_LEVEL_TAGS.TRACE
+        }
+      },
+      {
+        name: 'insecureConnect false',
+        options:
+        {
+          insecureConnect: false
+        }
+      },
+      {
+        name: 'insecureConnect true',
+        options:
+        {
+          insecureConnect: true
+        }
+      },
+      {
+        name: 'ocspFailOpen false',
+        options:
+        {
+          ocspFailOpen: false
+        }
+      },
+      {
+        name: 'ocspFailOpen true',
+        options:
+        {
+          ocspFailOpen: true
+        }
+      },
+      {
+        name: 'json parser',
+        options:
+        {
+          jsonColumnVariantParser: rawColumnValue => require('vm').runInNewContext("(" + rawColumnValue + ")")
+        }
+      },
+      {
+        name: 'xml parser',
+        options:
+        {
+          xmlColumnVariantParser: rawColumnValue => (require("fast-xml-parser").XMLParser()).parse(rawColumnValue)
```

Comment:
is `new` missing when XMLParser is created? here we give example where exception is probably thrown by who would catch it in our production code when variant is parsed?

Reply:
You're right, it was missing. Added `new` when creating XMLParser and corrected the parenthesis placement

I mentioned in a comment above about wrapping the xml parsing in its own try/catch if possible